pallets/payment-streams/src/utils.rs

```diff
@@ -550,180 +552,389 @@ where
+                            .rate
+                            .checked_mul(&time_passed_balance_typed)
+                            .ok_or(Error::<T>::ChargeOverflow)?;
+
+                        // Check the free balance of the user
+                        let user_balance = T::NativeBalance::reducible_balance(
+                            &user_account,
+                            Preservation::Preserve,
+                            Fortitude::Polite,
+                        );
+
+                        // If the user does not have enough balance to pay for its storage:
+                        if user_balance < amount_to_charge {
+                            // Check if this payment stream was already flagged as without funds and, if so, how many ticks have passed since then
+                            let out_of_funds_tick = fixed_rate_payment_stream.out_of_funds_tick;
+                            let current_tick = OnPollTicker::<T>::get();
+                            let ticks_since_out_of_funds = match out_of_funds_tick {
+                                Some(tick) => current_tick.saturating_sub(tick),
+                                None => {
+                                    // If this payment stream wasn't flagged as without funds before, flag it now with the current tick
+                                    FixedRatePaymentStreams::<T>::mutate(
+                                        provider_id,
+                                        user_account,
+                                        |payment_stream| {
+                                            let payment_stream = expect_or_err!(
+												payment_stream,
+												"Payment stream should exist if it was found before.",
+												Error::<T>::PaymentStreamNotFound
+											);
+                                            payment_stream.out_of_funds_tick = Some(current_tick);
+                                            Ok::<(), DispatchError>(())
+                                        },
+                                    )?;
+                                    Zero::zero()
+                                }
+                            };
+
+                            // If the user has missed payment for more than the allowed number of ticks (which is the number of ticks
+                            // that the user deposited when it was created), consider it as without funds.
+                            if ticks_since_out_of_funds >= T::NewStreamDeposit::get() {
+                                Self::manage_user_without_funds(
+                                    &provider_id,
+                                    &user_account,
+                                    &PaymentStream::FixedRatePaymentStream(
+                                        fixed_rate_payment_stream,
+                                    ),
+                                )?;
+                            }
```

Comment:
Why not inserting this user as without funds here?

Reply:
The user is inserted as without funds in the `manage_user_without_funds` function